src/main/java/build/buildfarm/worker/Slots.java

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm.worker;
+
+import com.google.common.collect.Sets;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Semaphore;
+
+public class Slots {
+  // Each stage has an input context queue.  These contexts are evaluated to become jobs that take
+  // claims.
+  public BlockingQueue<OperationContext> intake = new ArrayBlockingQueue<>(1);
```

Comment:
This still makes me queasy - I wish there was a better representation of our interchange here.

Reply:
I couldn't think of a better construct either.  I also wonder if anything > 1 would make a meaningful difference in behavior